diff --git a/src/build/GhosttyLib.zig b/src/build/GhosttyLib.zig
index b244a72c5..c5ff4405b 100644
--- a/src/build/GhosttyLib.zig
+++ b/src/build/GhosttyLib.zig
@@ -27,6 +27,7 @@ pub fn initStatic(
             .strip = deps.config.strip,
             .omit_frame_pointer = deps.config.strip,
             .unwind_tables = if (deps.config.strip) .none else .sync,
+            .pic = true,
         }),
         .linkage = .static,
     });
@@ -78,6 +79,7 @@ pub fn initShared(
             .strip = deps.config.strip,
             .omit_frame_pointer = deps.config.strip,
             .unwind_tables = if (deps.config.strip) .none else .sync,
+            .pic = true,
         }),
     });
     _ = try deps.add(lib);
