From 92c3286505282dc795ffc9a640e0762d62dd8e4a Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Mon, 3 Nov 2025 08:21:06 +0800
Subject: [PATCH] build for android

---
 Herebyfile.mjs | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/Herebyfile.mjs b/Herebyfile.mjs
index 7bbe76c1e..39f9310b7 100644
--- a/Herebyfile.mjs
+++ b/Herebyfile.mjs
@@ -1550,9 +1550,10 @@ function nodeToGOARCH(arch, os) {
 }
 
 const getPlatforms = memoize(() => {
-    let supportedPlatforms = publishAsTypescript
-        ? platforms
-        : platforms.filter(({ vsix }) => vsix);
+	/** @type {[os: OS, arch: Arch, cert: Cert, alpine?: boolean][]} */
+	let supportedPlatforms = [
+		{ os: "android", arch: "arm64", cert: "LinuxSign" },
+	];
 
     if (!options.forRelease) {
         supportedPlatforms = supportedPlatforms.filter(({ os, arch }) => os === process.platform && arch === process.arch);

From e6d423f03bc0590b390ce14d51d8828f4f9e4e4e Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Mon, 3 Nov 2025 09:16:51 +0800
Subject: [PATCH] remove specifiers

---
 Herebyfile.mjs | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Herebyfile.mjs b/Herebyfile.mjs
index 39f9310b7..c3c908831 100644
--- a/Herebyfile.mjs
+++ b/Herebyfile.mjs
@@ -1839,8 +1839,6 @@ async function runBuildNativePreviewPackages() {
             imports: undefined,
             dependencies: undefined,
             name: npmPackageName,
-            os: [nodeOs],
-            cpu: [nodeArch],
             exports: {
                 "./package.json": "./package.json",
             },
