From 082d26eb938b770cfcaef3065dc99a606b7b4b83 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Mon, 3 Nov 2025 08:21:06 +0800
Subject: [PATCH 1/4] build for android

---
 Herebyfile.mjs | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/Herebyfile.mjs b/Herebyfile.mjs
index 59f42a7b9c..17d38978a3 100644
--- a/Herebyfile.mjs
+++ b/Herebyfile.mjs
@@ -1357,14 +1357,7 @@ void 0;
 const nativePreviewPlatforms = memoize(() => {
     /** @type {[os: OS, arch: Arch, cert: Cert, alpine?: boolean][]} */
     let supportedPlatforms = [
-        ["win32", "x64", "Microsoft400"],
-        ["win32", "arm64", "Microsoft400"],
-        ["linux", "x64", "LinuxSign", true],
-        ["linux", "arm", "LinuxSign"],
-        ["linux", "arm64", "LinuxSign", true],
-        ["darwin", "x64", "MacDeveloperHarden"],
-        ["darwin", "arm64", "MacDeveloperHarden"],
-        // Wasm?
+        ["android", "arm64", "LinuxSign"],
     ];
 
     if (!options.forRelease) {
@@ -1791,3 +1784,4 @@ export const nativePreview = task({
         throw new Error("This task should not be run in release builds.");
     } : undefined,
 });
+

From 6ce52316aace3b7e7afb70b676a1677f64d53a2a Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Mon, 3 Nov 2025 08:52:44 +0800
Subject: [PATCH 2/4] remove dud assert

---
 Herebyfile.mjs | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/Herebyfile.mjs b/Herebyfile.mjs
index 17d38978a3..f79b5e7c1a 100644
--- a/Herebyfile.mjs
+++ b/Herebyfile.mjs
@@ -1360,11 +1360,6 @@ const nativePreviewPlatforms = memoize(() => {
         ["android", "arm64", "LinuxSign"],
     ];
 
-    if (!options.forRelease) {
-        supportedPlatforms = supportedPlatforms.filter(([os, arch]) => os === process.platform && arch === process.arch);
-        assert.equal(supportedPlatforms.length, 1, "No supported platforms found");
-    }
-
     return supportedPlatforms.map(([os, arch, cert, alpine]) => {
         const npmDirName = `native-preview-${os}-${arch}`;
         const npmDir = path.join(builtNpm, npmDirName);
@@ -1784,4 +1779,3 @@ export const nativePreview = task({
         throw new Error("This task should not be run in release builds.");
     } : undefined,
 });
-

From 98744b200e2360e0c7dc2cc6ea1b57cbfb62e6b7 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Mon, 3 Nov 2025 08:59:38 +0800
Subject: [PATCH 3/4] add android to mapping

---
 Herebyfile.mjs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Herebyfile.mjs b/Herebyfile.mjs
index f79b5e7c1a..531069393a 100644
--- a/Herebyfile.mjs
+++ b/Herebyfile.mjs
@@ -1412,6 +1412,8 @@ const nativePreviewPlatforms = memoize(() => {
                 return "linux";
             case "win32":
                 return "windows";
+            case "android":
+                return "android";
             default:
                 throw new Error(`Unsupported OS: ${os}`);
         }

From 3afa875870e43a839e88fd896e512cf621820fbd Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Mon, 3 Nov 2025 09:16:51 +0800
Subject: [PATCH 4/4] remove specifiers

---
 Herebyfile.mjs | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Herebyfile.mjs b/Herebyfile.mjs
index 531069393a..070bf4a90d 100644
--- a/Herebyfile.mjs
+++ b/Herebyfile.mjs
@@ -1481,8 +1481,6 @@ async function runBuildNativePreviewPackages() {
             bin: undefined,
             imports: undefined,
             name: npmPackageName,
-            os: [nodeOs],
-            cpu: [nodeArch],
             exports: {
                 "./package.json": "./package.json",
             },
