diff --git a/packages/core/scripts/build.ts b/packages/core/scripts/build.ts
index 2cd861fd..09f4029f 100644
--- a/packages/core/scripts/build.ts
+++ b/packages/core/scripts/build.ts
@@ -44,12 +44,7 @@ const buildAll = args.includes("--all") // Build for all platforms
 const gpaSafeStats = args.includes("--gpa-safe-stats")
 
 const variants: Variant[] = [
-  { platform: "darwin", arch: "x64" },
-  { platform: "darwin", arch: "arm64" },
-  { platform: "linux", arch: "x64" },
-  { platform: "linux", arch: "arm64" },
-  { platform: "win32", arch: "x64" },
-  { platform: "win32", arch: "arm64" },
+  { platform: "android", arch: "arm64" },
 ]
 
 const getHostVariant = (): Variant => {
@@ -75,9 +70,9 @@ const getZigTarget = (platform: string, arch: string): string => {
 const replaceLinks = (text: string): string => {
   return packageJson.homepage
     ? text.replace(
-        /(\[.*?\]\()(\.\/.*?\))/g,
-        (_, p1: string, p2: string) => `${p1}${packageJson.homepage}/blob/HEAD/${p2.replace("./", "")}`,
-      )
+      /(\[.*?\]\()(\.\/.*?\))/g,
+      (_, p1: string, p2: string) => `${p1}${packageJson.homepage}/blob/HEAD/${p2.replace("./", "")}`,
+    )
     : text
 }
 
@@ -91,13 +86,14 @@ if (missingRequired.length > 0) {
 if (buildNative) {
   console.log(`Building native ${isDev ? "dev" : "prod"} binaries${buildAll ? " for all platforms" : ""}...`)
 
-  const zigArgs = ["build", `-Doptimize=${isDev ? "Debug" : "ReleaseFast"}`]
+  const zigArgs = ["build", `-Doptimize=${isDev ? "Debug" : "ReleaseFast"}`, '--target=' + getZigTarget(variants[0].platform, variants[0].arch)]
   if (buildAll) {
     zigArgs.push("-Dall")
   }
   if (gpaSafeStats) {
     zigArgs.push("-Dgpa-safe-stats=true")
   }
+  console.log("Running Zig build with args:", zigArgs.join(" "))
 
   const zigBuild: SpawnSyncReturns<Buffer> = spawnSync("zig", zigArgs, {
     cwd: join(rootDir, "src", "zig"),
@@ -113,8 +109,11 @@ if (buildNative) {
     console.error("Error: Zig build failed")
     process.exit(1)
   }
+  console.log("Zig build completed successfully")
 
-  const variantsToPackage = buildAll ? variants : [getHostVariant()]
+  const variantsToPackage = [
+    { platform: "android", arch: "arm64" },
+  ];
 
   for (const { platform, arch } of variantsToPackage) {
     const nativeName = `${packageJson.name}-${platform}-${arch}`
