diff --git a/packages/core/scripts/build.ts b/packages/core/scripts/build.ts
index 2cd861fd..9feab68d 100644
--- a/packages/core/scripts/build.ts
+++ b/packages/core/scripts/build.ts
@@ -44,12 +44,7 @@ const buildAll = args.includes("--all") // Build for all platforms
 const gpaSafeStats = args.includes("--gpa-safe-stats")
 
 const variants: Variant[] = [
-  { platform: "darwin", arch: "x64" },
-  { platform: "darwin", arch: "arm64" },
-  { platform: "linux", arch: "x64" },
-  { platform: "linux", arch: "arm64" },
-  { platform: "win32", arch: "x64" },
-  { platform: "win32", arch: "arm64" },
+  { platform: "android", arch: "arm64" },
 ]
 
 const getHostVariant = (): Variant => {
@@ -75,9 +70,9 @@ const getZigTarget = (platform: string, arch: string): string => {
 const replaceLinks = (text: string): string => {
   return packageJson.homepage
     ? text.replace(
-        /(\[.*?\]\()(\.\/.*?\))/g,
-        (_, p1: string, p2: string) => `${p1}${packageJson.homepage}/blob/HEAD/${p2.replace("./", "")}`,
-      )
+      /(\[.*?\]\()(\.\/.*?\))/g,
+      (_, p1: string, p2: string) => `${p1}${packageJson.homepage}/blob/HEAD/${p2.replace("./", "")}`,
+    )
     : text
 }
 
@@ -114,7 +109,9 @@ if (buildNative) {
     process.exit(1)
   }
 
-  const variantsToPackage = buildAll ? variants : [getHostVariant()]
+  const variantsToPackage = [
+    { platform: "android", arch: "arm64" },
+  ];
 
   for (const { platform, arch } of variantsToPackage) {
     const nativeName = `${packageJson.name}-${platform}-${arch}`
