commit 81d14742804a8df40c1c24b0cfdc853b03b96d9b
Author: Elliana May <me@mause.me>
Date:   Tue Mar 17 19:11:35 2026 +0800

    patch

diff --git a/crates/jsonschema-cli/Cargo.toml b/crates/jsonschema-cli/Cargo.toml
index a1ab5c6..38a9203 100644
--- a/crates/jsonschema-cli/Cargo.toml
+++ b/crates/jsonschema-cli/Cargo.toml
@@ -17,7 +17,9 @@ path = "src/main.rs"
 
 [dependencies]
 clap = { version = "4.5", features = ["derive"] }
-jsonschema = { version = "0.45.0", path = "../jsonschema/" }
+jsonschema = { path = "../jsonschema/", version = "0.45.0", default-features = false, features = [
+  "resolve-http", "resolve-file", "tls-ring", "resolve-async"
+] }
 percent-encoding = "2.3"
 referencing = { version = "0.45.0", path = "../jsonschema-referencing" }
 serde-saphyr = { version = "0.0.21", default-features = false }
diff --git a/crates/jsonschema/Cargo.toml b/crates/jsonschema/Cargo.toml
index c1649e8..7b38368 100644
--- a/crates/jsonschema/Cargo.toml
+++ b/crates/jsonschema/Cargo.toml
@@ -30,11 +30,11 @@ arbitrary-precision = ["serde_json/arbitrary_precision", "dep:num-bigint"]
 
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies]
 # Use rustls-no-provider to allow manual selection of crypto backend
-reqwest = { version = "0.13", default-features = false, features = [
+reqwest = { version = "0.12", default-features = false, features = [
   "blocking",
   "json",
   "http2",
-  "rustls-no-provider",
+  "rustls-tls-webpki-roots"
 ], optional = true }
 # Crypto provider for TLS: aws-lc-rs by default, or ring when tls-ring feature is enabled
 rustls = { version = "0.23", default-features = false, features = ["std", "tls12"], optional = true }
