From c39e610eeae8146f8c440ef26cbfe82eac6c4121 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Sun, 29 Mar 2026 17:55:30 +0800
Subject: [PATCH] add android platform

---
 BUILD.bazel | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/BUILD.bazel b/BUILD.bazel
index f9f8c1a37..a88b8c5de 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -63,6 +63,13 @@ platform(
 )
 
 # Used for cross-compilation
+platform(
+    name = "android_arm64",
+    constraint_values = [
+        "@platforms//os:android",
+        "@platforms//cpu:arm64",
+    ],
+)
 platform(
     name = "macOS_x86",
     constraint_values = [

From 81459cb42cef8bc8947b5eaf292c5e9cdc4d44bb Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Sun, 29 Mar 2026 18:19:49 +0800
Subject: [PATCH] steal some config from ink

---
 BUILD.bazel  | 32 +++++++++++++++++++++++++++++---
 MODULE.bazel | 13 +++++++++++++
 2 files changed, 42 insertions(+), 3 deletions(-)

diff --git a/BUILD.bazel b/BUILD.bazel
index a88b8c5de..9efddf8d7 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -64,17 +64,43 @@ platform(
 
 # Used for cross-compilation
 platform(
-    name = "android_arm64",
+    name = "macOS_x86",
+    constraint_values = [
+        "@platforms//os:macos",
+        "@platforms//cpu:x86_64",
+    ],
+)
+
+# Android platforms for building with NDK.
+platform(
+    name = "android-arm",
     constraint_values = [
+        "@platforms//cpu:armv7",
         "@platforms//os:android",
+    ],
+)
+
+platform(
+    name = "android-arm64",
+    constraint_values = [
         "@platforms//cpu:arm64",
+        "@platforms//os:android",
     ],
 )
+
 platform(
-    name = "macOS_x86",
+    name = "android-x86",
+    constraint_values = [
+        "@platforms//cpu:x86_32",
+        "@platforms//os:android",
+    ],
+)
+
+platform(
+    name = "android-x86_64",
     constraint_values = [
-        "@platforms//os:macos",
         "@platforms//cpu:x86_64",
+        "@platforms//os:android",
     ],
 )
 
diff --git a/MODULE.bazel b/MODULE.bazel
index 396a7dc41..4e24b26e4 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -46,3 +46,16 @@ use_repo(compat, "cc_compatibility_proxy")
 
 cc_configure = use_extension("@rules_cc//cc:extensions.bzl", "cc_configure_extension")
 use_repo(cc_configure, "local_config_cc")
+
+# Configure the Android NDK toolchain for NDK builds. Don't register that here so it's not required
+# for builds that don't need it. Those configs can register those in --extra_toolchains.
+bazel_dep(name = "rules_android_ndk", version = "0.1.3")
+
+android_ndk_repository_extension = use_extension(
+    "@rules_android_ndk//:extension.bzl",
+    "android_ndk_repository_extension",
+)
+use_repo(android_ndk_repository_extension, "androidndk")
+
+bazel_dep(name = "rules_android", version = "0.6.0")
+bazel_dep(name = "rules_go", version = "0.60.0")

From 66f12be10bed1fa00ee68d98db7e81a8b50b381a Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Sun, 29 Mar 2026 22:18:32 +0800
Subject: [PATCH] add android target triple

---
 build/deps/rust.MODULE.bazel | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build/deps/rust.MODULE.bazel b/build/deps/rust.MODULE.bazel
index 4e4506dad..2e1c23f48 100644
--- a/build/deps/rust.MODULE.bazel
+++ b/build/deps/rust.MODULE.bazel
@@ -8,6 +8,7 @@ RUST_TARGET_TRIPLES = [
     "x86_64-apple-darwin",
     # Add support for macOS rosetta
     "aarch64-unknown-linux-gnu",
+    "aarch64-linux-android",
 ]
 
 # rules_rust

From 023362365b66ba66a30e8d662d9f5f8c0c2aca94 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Sun, 29 Mar 2026 22:57:29 +0800
Subject: [PATCH] add patch

---
 build/deps/deps.jsonc             |  5 ++++-
 build/deps/gen/deps.MODULE.bazel  |  1 +
 patches/zlib/0002-Add-dummy.patch | 22 ++++++++++++++++++++++
 3 files changed, 27 insertions(+), 1 deletion(-)
 create mode 100644 patches/zlib/0002-Add-dummy.patch

diff --git a/build/deps/deps.jsonc b/build/deps/deps.jsonc
index 73b37cec3..dbd4cfcab 100644
--- a/build/deps/deps.jsonc
+++ b/build/deps/deps.jsonc
@@ -124,7 +124,10 @@
       "url": "https://chromium.googlesource.com/chromium/src/third_party/zlib.git",
       "branch": "main",
       "build_file": "//:build/BUILD.zlib",
-      "patches": ["//:patches/zlib/0001-Add-dummy-MODULE.bazel.patch"]
+      "patches": [
+        "//:patches/zlib/0001-Add-dummy-MODULE.bazel.patch",
+        "//:patches/zlib/0002-Add-dummy.patch"
+      ]
     },
     {
       "name": "workerd-cxx",
diff --git a/build/deps/gen/deps.MODULE.bazel b/build/deps/gen/deps.MODULE.bazel
index de267adad..69075b7cd 100644
--- a/build/deps/gen/deps.MODULE.bazel
+++ b/build/deps/gen/deps.MODULE.bazel
@@ -152,6 +152,7 @@ git_override(
     patch_strip = 1,
     patches = [
         "//:patches/zlib/0001-Add-dummy-MODULE.bazel.patch",
+        "//:patches/zlib/0002-Add-dummy.patch",
     ],
     remote = "https://chromium.googlesource.com/chromium/src/third_party/zlib.git",
 )
diff --git a/patches/zlib/0002-Add-dummy.patch b/patches/zlib/0002-Add-dummy.patch
new file mode 100644
index 000000000..f8d3f302e
--- /dev/null
+++ b/patches/zlib/0002-Add-dummy.patch
@@ -0,0 +1,22 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Felix Hanau <felix@cloudflare.com>
+Date: Sun, 8 Jun 2025 15:55:56 -0400
+Subject: Add dummy MODULE.bazel
+
+
+diff --git a/MODULE.bazel b/MODULE.bazel
+new file mode 100644
+index 0000000000000000000000000000000000000000..3611da399064ef179d4924a67c3bbdda5a765d15
+--- /dev/null
++++ b/MODULE.bazel
+@@ -0,0 +1,10 @@
++# dummy file used to support bzlmod
++module(
++    name = "zlib",
++    version = "0.0.0",
++    compatibility_level = 1,
++)
++
++bazel_dep(name = "platforms", version = "1.0.0")
++bazel_dep(name = "rules_cc", version = "0.1.1")
++bazel_dep(name = "bazel_skylib", version = "1.7.1")

From 56952df45997f5bd0718768249193a431f84b250 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Sun, 29 Mar 2026 23:06:32 +0800
Subject: [PATCH] tweak platform name

---
 BUILD.bazel | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/BUILD.bazel b/BUILD.bazel
index 9efddf8d7..bbf4ebf90 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -81,7 +81,7 @@ platform(
 )
 
 platform(
-    name = "android-arm64",
+    name = "android-aarch64",
     constraint_values = [
         "@platforms//cpu:arm64",
         "@platforms//os:android",
