commit 401d7b75d72c86bc86c961a04639df8305ca918b
Author: Elliana May <me@mause.me>
Date:   Sat Mar 14 14:14:34 2026 +0800

    target

diff --git a/packages/cli/build.ts b/packages/cli/build.ts
index d474f5d2..91854c17 100644
--- a/packages/cli/build.ts
+++ b/packages/cli/build.ts
@@ -102,6 +102,10 @@ async function buildNapiBinding() {
 
   const { task } = await cli.build({
     ...passedInOptions,
+		target: process.env.CARGO_TARGET_NAME || (() => {
+			throw new Error('CARGO_TARGET_NAME environment variable is not set. Make sure to run this script via npm scripts that set this variable, or set it manually when running the script directly.'));
+	})()
+		,
     packageJsonPath: '../package.json',
     cwd: 'binding',
     platform: true,
