diff --git a/crates/uv-distribution-filename/src/wheel.rs b/crates/uv-distribution-filename/src/wheel.rs
index c256931e4..61554287b 100644
--- a/crates/uv-distribution-filename/src/wheel.rs
+++ b/crates/uv-distribution-filename/src/wheel.rs
@@ -87,6 +87,10 @@ impl WheelFilename {
 
     /// Returns `true` if the wheel is compatible with the given tags.
     pub fn is_compatible(&self, compatible_tags: &Tags) -> bool {
+        println!("platform_tags: {:?}", self.platform_tags());
+        println!("compatible_tags: {:?}", compatible_tags);
+        println!("python_tags: {:?}", self.python_tags());
+        println!("abi_tags: {:?}", self.abi_tags());
         compatible_tags.is_compatible(self.python_tags(), self.abi_tags(), self.platform_tags())
     }
 
diff --git a/crates/uv-distribution-types/src/installed.rs b/crates/uv-distribution-types/src/installed.rs
index 1f6a0f9d0..ac0a78f80 100644
--- a/crates/uv-distribution-types/src/installed.rs
+++ b/crates/uv-distribution-types/src/installed.rs
@@ -504,7 +504,14 @@ impl InstalledDist {
 
         // Parse the tags.
         let tags = if let Some(tags) = wheel_file.tags() {
-            Some(ExpandedTags::parse(tags.iter().map(String::as_str))?)
+            warn!(
+                "Using tags: {tags:?} from WHEEL file at `{path:?}`. This is not guaranteed to be correct, and may be missing or inaccurate. Consider reinstalling the package with a modern installer that records accurate tags in the WHEEL file."
+            );
+            Some(ExpandedTags::parse(vec![
+                "android_24_arm64_v8a",
+                "android_24_aarch64",
+                "android_35_arm64_v8a",
+            ])?)
         } else {
             None
         };
diff --git a/crates/uv-installer/src/satisfies.rs b/crates/uv-installer/src/satisfies.rs
index a3d8c0c62..8efe9ca1f 100644
--- a/crates/uv-installer/src/satisfies.rs
+++ b/crates/uv-installer/src/satisfies.rs
@@ -5,6 +5,7 @@ use same_file::is_same_file;
 use tracing::{debug, trace};
 use url::Url;
 
+use tracing::warn;
 use uv_cache_info::CacheInfo;
 use uv_cache_key::{CanonicalUrl, RepositoryUrl};
 use uv_distribution_filename::ExpandedTags;
@@ -350,6 +351,7 @@ impl RequirementSatisfaction {
 
         // If the distribution isn't compatible with the current platform, it is a mismatch.
         if let Ok(Some(wheel_tags)) = distribution.read_tags() {
+            warn!("{distribution} tags: {wheel_tags:?}, current tags: {tags:?}");
             if !wheel_tags.is_compatible(tags) {
                 if let Some(hint) = generate_dist_compatibility_hint(wheel_tags, tags) {
                     debug!("Platform tags mismatch for {distribution}: {hint}");
