diff --git a/crates/uv-platform/src/lib.rs b/crates/uv-platform/src/lib.rs
index d0e6fdd4b..e5403204b 100644
--- a/crates/uv-platform/src/lib.rs
+++ b/crates/uv-platform/src/lib.rs
@@ -18,7 +18,7 @@ mod os;
 
 #[derive(Error, Debug)]
 pub enum Error {
-    #[error("Unknown operating system: {0}")]
+    #[error("uv-platform Unknown operating system: {0}")]
     UnknownOs(String),
     #[error("Unknown architecture: {0}")]
     UnknownArch(String),
diff --git a/crates/uv-python/python/get_interpreter_info.py b/crates/uv-python/python/get_interpreter_info.py
index ca83e8d8d..95c2990a8 100644
--- a/crates/uv-python/python/get_interpreter_info.py
+++ b/crates/uv-python/python/get_interpreter_info.py
@@ -446,7 +446,7 @@ def get_operating_system_and_architecture():
         )
         sys.exit(0)
 
-    if operating_system == "linux":
+    if operating_system == "linux" or operating_system == "android":
         # noinspection PyProtectedMember
         from .packaging._manylinux import _get_glibc_version
 
diff --git a/crates/uv-python/src/discovery.rs b/crates/uv-python/src/discovery.rs
index 29659116c..2e1cf76ad 100644
--- a/crates/uv-python/src/discovery.rs
+++ b/crates/uv-python/src/discovery.rs
@@ -1011,14 +1011,14 @@ impl Error {
                 InterpreterError::UnexpectedResponse(UnexpectedResponseError { path, .. })
                 | InterpreterError::StatusCode(StatusCodeError { path, .. }) => {
                     debug!(
-                        "Skipping bad interpreter at {} from {source}: {err}",
+                        "StatusCode Skipping bad interpreter at {} from {source}: {err}",
                         path.display()
                     );
                     false
                 }
                 InterpreterError::QueryScript { path, err } => {
                     debug!(
-                        "Skipping bad interpreter at {} from {source}: {err}",
+                        "QueryScript Skipping bad interpreter at {} from {source}: {err}",
                         path.display()
                     );
                     false
@@ -1026,7 +1026,7 @@ impl Error {
                 #[cfg(windows)]
                 InterpreterError::CorruptWindowsPackage { path, err } => {
                     debug!(
-                        "Skipping bad interpreter at {} from {source}: {err}",
+                        "CorruptWindowsPackage Skipping bad interpreter at {} from {source}: {err}",
                         path.display()
                     );
                     false
diff --git a/crates/uv-python/src/interpreter.rs b/crates/uv-python/src/interpreter.rs
index 6bde8ddad..36345cca8 100644
--- a/crates/uv-python/src/interpreter.rs
+++ b/crates/uv-python/src/interpreter.rs
@@ -888,7 +888,7 @@ pub enum InterpreterInfoError {
         "Broken Python installation, `platform.mac_ver()` returned an empty value, please reinstall Python"
     )]
     BrokenMacVer,
-    #[error("Unknown operating system: `{operating_system}`")]
+    #[error("uv-python Unknown operating system: `{operating_system}`")]
     UnknownOperatingSystem { operating_system: String },
     #[error("Python {python_version} is not supported. Please use Python 3.8 or newer.")]
     UnsupportedPythonVersion { python_version: String },
