diff --git a/extension/extension_config_local.cmake b/extension/extension_config_local.cmake
new file mode 100644
index 0000000000..40fea8a3be
--- /dev/null
+++ b/extension/extension_config_local.cmake
@@ -0,0 +1,8 @@
+duckdb_extension_load(ui
+    GIT_URL https://github.com/duckdb/duckdb-ui
+    GIT_TAG c1f6be62e3cc01719da4c2c4a8287d28661ba5bc
+)
+#duckdb_extension_load(delta
+#    GIT_URL https://github.com/duckdb/duckdb-delta
+#    GIT_TAG ef8ba4d0e8c89d403c3011e276531fdc56bfcd36
+#)
