From 33d964b66bf826e2db0999a2ad302fc6f9c83669 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Fri, 21 Nov 2025 14:03:46 +0800
Subject: [PATCH] fix: add other typescript extensions

---
 src/syntax/languages/mod.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/syntax/languages/mod.rs b/src/syntax/languages/mod.rs
index 84f85ff..978dec0 100644
--- a/src/syntax/languages/mod.rs
+++ b/src/syntax/languages/mod.rs
@@ -34,7 +34,7 @@ pub fn get_language(path: &Path) -> Option<(Language, &'static str)> {
 
     match extension {
         "rs" => Some((rust::language(), rust::HIGHLIGHT_QUERY)),
-        "ts" | "tsx" => Some((typescript::language(), typescript::HIGHLIGHT_QUERY)),
+        "ts" | "tsx" | "mts" | "cts" => Some((typescript::language(), typescript::HIGHLIGHT_QUERY)),
         "js" | "jsx" | "mjs" | "cjs" => Some((javascript::language(), javascript::HIGHLIGHT_QUERY)),
         "py" | "pyw" => Some((python::language(), python::HIGHLIGHT_QUERY)),
         "sh" | "bash" | "zsh" => Some((bash::language(), bash::HIGHLIGHT_QUERY)),
