From 220dad0ccced8f70480ed4a13e0fe7fea3eec258 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 14:50:37 +0800
Subject: [PATCH] override rubyhdrdir

---
 ext/nokogiri/extconf.rb | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ext/nokogiri/extconf.rb b/ext/nokogiri/extconf.rb
index 04ffc9fa..34e6ef00 100644
--- a/ext/nokogiri/extconf.rb
+++ b/ext/nokogiri/extconf.rb
@@ -3,6 +3,7 @@
 # rubocop:disable Style/GlobalVars
 
 ENV["RC_ARCHS"] = "" if RUBY_PLATFORM.include?("darwin")
+RbConfig::CONFIG["rubyhdrdir"] = File.join(ENV["TERMUX_PREFIX"], "include", "ruby-#{RbConfig::CONFIG['ruby_version']}")
 
 require "mkmf"
 require "rbconfig"

From be4b346980dccf544b934c25d94d6bc900387bb1 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 14:52:20 +0800
Subject: [PATCH] use patch sed

---
 ext/nokogiri/extconf.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/nokogiri/extconf.rb b/ext/nokogiri/extconf.rb
index 34e6ef00..2671b180 100644
--- a/ext/nokogiri/extconf.rb
+++ b/ext/nokogiri/extconf.rb
@@ -3,7 +3,7 @@
 # rubocop:disable Style/GlobalVars
 
 ENV["RC_ARCHS"] = "" if RUBY_PLATFORM.include?("darwin")
-RbConfig::CONFIG["rubyhdrdir"] = File.join(ENV["TERMUX_PREFIX"], "include", "ruby-#{RbConfig::CONFIG['ruby_version']}")
+RbConfig::CONFIG["rubyhdrdir"] = File.join("@TERMUX_PREFIX@", "include", "ruby-#{RbConfig::CONFIG['ruby_version']}")
 
 require "mkmf"
 require "rbconfig"

From 92a24fdffc8b127b64cb842bb20e03010363ecdf Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 15:40:59 +0800
Subject: [PATCH] dump

---
 ext/nokogiri/extconf.rb | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/ext/nokogiri/extconf.rb b/ext/nokogiri/extconf.rb
index 2671b180..04de2810 100644
--- a/ext/nokogiri/extconf.rb
+++ b/ext/nokogiri/extconf.rb
@@ -1149,6 +1149,22 @@ unless config_system_libraries?
   end
 end
 
+puts "Configuration summary:"
+puts "  - using system libraries: #{config_system_libraries?}"
+puts "  - using static linking: #{static_p}"
+puts "  - using cross build: #{cross_build_p}"
+puts "  - other library versions: #{other_library_versions_string}"
+puts "  - cppflags: #{$CPPFLAGS}"
+puts "  - ldflags: #{$LDFLAGS}"
+puts "  - libs: #{$libs}"
+puts "  - incflags: #{$INCFLAGS}"
+puts "  - archflags: #{RbConfig::CONFIG["ARCH_FLAG"]}"
+puts "  - ruby version: #{RbConfig::CONFIG["ruby_version"]}"
+puts "  - ruby platform: #{RbConfig::CONFIG["ruby_platform"]}"
+puts "  - ruby host: #{RbConfig::CONFIG["host"]}"
+puts "  - ruby target: #{RbConfig::CONFIG["target"]}"
+puts "  - ruby config: #{RbConfig::CONFIG.inspect}"
+
 create_makefile("nokogiri/nokogiri")
 
 if config_clean?

From 97e87dc2379145faf55a9b71a1b729ccba01ded0 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 16:07:33 +0800
Subject: [PATCH] more dump

---
 ext/nokogiri/extconf.rb | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ext/nokogiri/extconf.rb b/ext/nokogiri/extconf.rb
index 04de2810..c6afc1a8 100644
--- a/ext/nokogiri/extconf.rb
+++ b/ext/nokogiri/extconf.rb
@@ -4,6 +4,7 @@
 
 ENV["RC_ARCHS"] = "" if RUBY_PLATFORM.include?("darwin")
 RbConfig::CONFIG["rubyhdrdir"] = File.join("@TERMUX_PREFIX@", "include", "ruby-#{RbConfig::CONFIG['ruby_version']}")
+puts "Using rubyhdrdir=#{RbConfig::CONFIG['rubyhdrdir']}"
 
 require "mkmf"
 require "rbconfig"

From d0c66fd7f53b08e9a5b6037c573387b9a8e41552 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 16:38:32 +0800
Subject: [PATCH] try something stupid

---
 rbconfig.rb | 2 ++
 1 file changed, 2 insertions(+)
 create mode 100644 rbconfig.rb

diff --git a/rbconfig.rb b/rbconfig.rb
new file mode 100644
index 00000000..5bf0e54d
--- /dev/null
+++ b/rbconfig.rb
@@ -0,0 +1,2 @@
+require 'rbconfig'
+puts 'aaaaaaaaaaaaaa'

From 0fabb41a0d1cbd6b5adf08ffa68c197f03f0df79 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 16:49:51 +0800
Subject: [PATCH] intercept

---
 rbconfig-patch.rb | 4 ++++
 rbconfig.rb       | 2 --
 2 files changed, 4 insertions(+), 2 deletions(-)
 create mode 100644 rbconfig-patch.rb
 delete mode 100644 rbconfig.rb

diff --git a/rbconfig-patch.rb b/rbconfig-patch.rb
new file mode 100644
index 00000000..0a4d372d
--- /dev/null
+++ b/rbconfig-patch.rb
@@ -0,0 +1,4 @@
+require 'rbconfig'
+
+RbConfig::CONFIG["rubyhdrdir"] = File.join("@TERMUX_PREFIX@", "include", "ruby-#{RbConfig::CONFIG['ruby_version']}")
+puts "Using rubyhdrdir=#{RbConfig::CONFIG['rubyhdrdir']}"
diff --git a/rbconfig.rb b/rbconfig.rb
deleted file mode 100644
index 5bf0e54d..00000000
--- a/rbconfig.rb
+++ /dev/null
@@ -1,2 +0,0 @@
-require 'rbconfig'
-puts 'aaaaaaaaaaaaaa'

From 6d30aa4f32845698eee18a1c55e5234e3cd84976 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 17:01:05 +0800
Subject: [PATCH] sanity

---
 rbconfig-patch.rb | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/rbconfig-patch.rb b/rbconfig-patch.rb
index 0a4d372d..ade6cc7c 100644
--- a/rbconfig-patch.rb
+++ b/rbconfig-patch.rb
@@ -2,3 +2,6 @@ require 'rbconfig'
 
 RbConfig::CONFIG["rubyhdrdir"] = File.join("@TERMUX_PREFIX@", "include", "ruby-#{RbConfig::CONFIG['ruby_version']}")
 puts "Using rubyhdrdir=#{RbConfig::CONFIG['rubyhdrdir']}"
+if not File.exist?(($hdrdir = RbConfig::CONFIG["rubyhdrdir"]) + "/ruby/ruby.h")
+  abort "#{hdrdir}/ruby/ruby.h does not exist"
+end

From 75a4a8f9bc46fe2f3a80819a58b682ed8b217eb5 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 17:10:37 +0800
Subject: [PATCH] fix var ref

---
 rbconfig-patch.rb | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/rbconfig-patch.rb b/rbconfig-patch.rb
index ade6cc7c..75956833 100644
--- a/rbconfig-patch.rb
+++ b/rbconfig-patch.rb
@@ -1,7 +1,8 @@
 require 'rbconfig'
 
 RbConfig::CONFIG["rubyhdrdir"] = File.join("@TERMUX_PREFIX@", "include", "ruby-#{RbConfig::CONFIG['ruby_version']}")
-puts "Using rubyhdrdir=#{RbConfig::CONFIG['rubyhdrdir']}"
-if not File.exist?(($hdrdir = RbConfig::CONFIG["rubyhdrdir"]) + "/ruby/ruby.h")
-  abort "#{hdrdir}/ruby/ruby.h does not exist"
+$hdrdir = RbConfig::CONFIG["rubyhdrdir"]
+puts "Using rubyhdrdir=#{$hdrdir}"
+if not File.exist?($hdrdir + "/ruby/ruby.h")
+  abort "#{$hdrdir}/ruby/ruby.h does not exist"
 end

From a3ce5660519c4af2b8d4b3afd49e3580af794b61 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 17:16:51 +0800
Subject: [PATCH] OG

---
 rbconfig-patch.rb | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/rbconfig-patch.rb b/rbconfig-patch.rb
index 75956833..89afd486 100644
--- a/rbconfig-patch.rb
+++ b/rbconfig-patch.rb
@@ -1,5 +1,7 @@
 require 'rbconfig'
 
+$hdrdir = RbConfig::CONFIG["rubyhdrdir"]
+puts "Original hdrdir: #{$hdrdir}"
 RbConfig::CONFIG["rubyhdrdir"] = File.join("@TERMUX_PREFIX@", "include", "ruby-#{RbConfig::CONFIG['ruby_version']}")
 $hdrdir = RbConfig::CONFIG["rubyhdrdir"]
 puts "Using rubyhdrdir=#{$hdrdir}"

From 5d620e2c8347e986e7b3ddd5169643f169ca565e Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 17:23:43 +0800
Subject: [PATCH] retarget

---
 rbconfig-patch.rb | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/rbconfig-patch.rb b/rbconfig-patch.rb
index 89afd486..ec94694d 100644
--- a/rbconfig-patch.rb
+++ b/rbconfig-patch.rb
@@ -5,6 +5,7 @@ puts "Original hdrdir: #{$hdrdir}"
 RbConfig::CONFIG["rubyhdrdir"] = File.join("@TERMUX_PREFIX@", "include", "ruby-#{RbConfig::CONFIG['ruby_version']}")
 $hdrdir = RbConfig::CONFIG["rubyhdrdir"]
 puts "Using rubyhdrdir=#{$hdrdir}"
-if not File.exist?($hdrdir + "/ruby/ruby.h")
-  abort "#{$hdrdir}/ruby/ruby.h does not exist"
+$target = $hdrdir + "/ruby.h"
+if not File.exist?($hdrdir + "/ruby.h")
+  abort "#{$hdrdir}/ruby.h does not exist"
 end

From 3faa157152df0507cf1e741252ac447b325ebbef Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 17:23:52 +0800
Subject: [PATCH] ls children

---
 rbconfig-patch.rb | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rbconfig-patch.rb b/rbconfig-patch.rb
index ec94694d..93623f0c 100644
--- a/rbconfig-patch.rb
+++ b/rbconfig-patch.rb
@@ -2,6 +2,7 @@ require 'rbconfig'
 
 $hdrdir = RbConfig::CONFIG["rubyhdrdir"]
 puts "Original hdrdir: #{$hdrdir}"
+puts Dir.children("@TERMUX_PREFIX@")
 RbConfig::CONFIG["rubyhdrdir"] = File.join("@TERMUX_PREFIX@", "include", "ruby-#{RbConfig::CONFIG['ruby_version']}")
 $hdrdir = RbConfig::CONFIG["rubyhdrdir"]
 puts "Using rubyhdrdir=#{$hdrdir}"

From bc6c5f20111d6add3e1eebdd120bc48382b73a93 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 17:27:12 +0800
Subject: [PATCH] dir hdrdir

---
 rbconfig-patch.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rbconfig-patch.rb b/rbconfig-patch.rb
index 93623f0c..1662a543 100644
--- a/rbconfig-patch.rb
+++ b/rbconfig-patch.rb
@@ -2,10 +2,10 @@ require 'rbconfig'
 
 $hdrdir = RbConfig::CONFIG["rubyhdrdir"]
 puts "Original hdrdir: #{$hdrdir}"
-puts Dir.children("@TERMUX_PREFIX@")
 RbConfig::CONFIG["rubyhdrdir"] = File.join("@TERMUX_PREFIX@", "include", "ruby-#{RbConfig::CONFIG['ruby_version']}")
 $hdrdir = RbConfig::CONFIG["rubyhdrdir"]
 puts "Using rubyhdrdir=#{$hdrdir}"
+puts Dir.children($hdrdir)
 $target = $hdrdir + "/ruby.h"
 if not File.exist?($hdrdir + "/ruby.h")
   abort "#{$hdrdir}/ruby.h does not exist"

From 0e3f79ceb40634b22ceef7532577015b05233e5d Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 17:30:16 +0800
Subject: [PATCH] dir include

---
 rbconfig-patch.rb | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rbconfig-patch.rb b/rbconfig-patch.rb
index 1662a543..bcd36834 100644
--- a/rbconfig-patch.rb
+++ b/rbconfig-patch.rb
@@ -5,6 +5,7 @@ puts "Original hdrdir: #{$hdrdir}"
 RbConfig::CONFIG["rubyhdrdir"] = File.join("@TERMUX_PREFIX@", "include", "ruby-#{RbConfig::CONFIG['ruby_version']}")
 $hdrdir = RbConfig::CONFIG["rubyhdrdir"]
 puts "Using rubyhdrdir=#{$hdrdir}"
+puts Dir.children(File.join("@TERMUX_PREFIX@", "include"))
 puts Dir.children($hdrdir)
 $target = $hdrdir + "/ruby.h"
 if not File.exist?($hdrdir + "/ruby.h")
