From 498358e03800db0067b63618111457169249dfc5 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 14:50:37 +0800
Subject: [PATCH] override rubyhdrdir

---
 ext/nokogiri/extconf.rb | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ext/nokogiri/extconf.rb b/ext/nokogiri/extconf.rb
index 04ffc9fa..34e6ef00 100644
--- a/ext/nokogiri/extconf.rb
+++ b/ext/nokogiri/extconf.rb
@@ -3,6 +3,7 @@
 # rubocop:disable Style/GlobalVars
 
 ENV["RC_ARCHS"] = "" if RUBY_PLATFORM.include?("darwin")
+RbConfig::CONFIG["rubyhdrdir"] = File.join(ENV["TERMUX_PREFIX"], "include", "ruby-#{RbConfig::CONFIG['ruby_version']}")
 
 require "mkmf"
 require "rbconfig"

From 2a0ed5e7e3762a43b475e8ab1fef2a3c9aa1189a Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 14:52:20 +0800
Subject: [PATCH] use patch sed

---
 ext/nokogiri/extconf.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/nokogiri/extconf.rb b/ext/nokogiri/extconf.rb
index 34e6ef00..2671b180 100644
--- a/ext/nokogiri/extconf.rb
+++ b/ext/nokogiri/extconf.rb
@@ -3,7 +3,7 @@
 # rubocop:disable Style/GlobalVars
 
 ENV["RC_ARCHS"] = "" if RUBY_PLATFORM.include?("darwin")
-RbConfig::CONFIG["rubyhdrdir"] = File.join(ENV["TERMUX_PREFIX"], "include", "ruby-#{RbConfig::CONFIG['ruby_version']}")
+RbConfig::CONFIG["rubyhdrdir"] = File.join("@TERMUX_PREFIX@", "include", "ruby-#{RbConfig::CONFIG['ruby_version']}")
 
 require "mkmf"
 require "rbconfig"

From f19ca6f632f9f14692a9660c184c9b8c70ac8057 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 15:40:59 +0800
Subject: [PATCH] dump

---
 ext/nokogiri/extconf.rb | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/ext/nokogiri/extconf.rb b/ext/nokogiri/extconf.rb
index 2671b180..04de2810 100644
--- a/ext/nokogiri/extconf.rb
+++ b/ext/nokogiri/extconf.rb
@@ -1149,6 +1149,22 @@ unless config_system_libraries?
   end
 end
 
+puts "Configuration summary:"
+puts "  - using system libraries: #{config_system_libraries?}"
+puts "  - using static linking: #{static_p}"
+puts "  - using cross build: #{cross_build_p}"
+puts "  - other library versions: #{other_library_versions_string}"
+puts "  - cppflags: #{$CPPFLAGS}"
+puts "  - ldflags: #{$LDFLAGS}"
+puts "  - libs: #{$libs}"
+puts "  - incflags: #{$INCFLAGS}"
+puts "  - archflags: #{RbConfig::CONFIG["ARCH_FLAG"]}"
+puts "  - ruby version: #{RbConfig::CONFIG["ruby_version"]}"
+puts "  - ruby platform: #{RbConfig::CONFIG["ruby_platform"]}"
+puts "  - ruby host: #{RbConfig::CONFIG["host"]}"
+puts "  - ruby target: #{RbConfig::CONFIG["target"]}"
+puts "  - ruby config: #{RbConfig::CONFIG.inspect}"
+
 create_makefile("nokogiri/nokogiri")
 
 if config_clean?

From dc0d87384b2061a6cbcf0883c2ec3520d4293352 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 16:07:33 +0800
Subject: [PATCH] more dump

---
 ext/nokogiri/extconf.rb | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ext/nokogiri/extconf.rb b/ext/nokogiri/extconf.rb
index 04de2810..c6afc1a8 100644
--- a/ext/nokogiri/extconf.rb
+++ b/ext/nokogiri/extconf.rb
@@ -4,6 +4,7 @@
 
 ENV["RC_ARCHS"] = "" if RUBY_PLATFORM.include?("darwin")
 RbConfig::CONFIG["rubyhdrdir"] = File.join("@TERMUX_PREFIX@", "include", "ruby-#{RbConfig::CONFIG['ruby_version']}")
+puts "Using rubyhdrdir=#{RbConfig::CONFIG['rubyhdrdir']}"
 
 require "mkmf"
 require "rbconfig"

From 4912c7abf37bee02b18211412cbffeb22c997628 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 16:38:32 +0800
Subject: [PATCH] try something stupid

---
 rbconfig.rb | 2 ++
 1 file changed, 2 insertions(+)
 create mode 100644 rbconfig.rb

diff --git a/rbconfig.rb b/rbconfig.rb
new file mode 100644
index 00000000..5bf0e54d
--- /dev/null
+++ b/rbconfig.rb
@@ -0,0 +1,2 @@
+require 'rbconfig'
+puts 'aaaaaaaaaaaaaa'

From 3ed311ce2b2cde80442745777feb15343cc91bff Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 16:49:51 +0800
Subject: [PATCH] intercept

---
 rbconfig-patch.rb | 4 ++++
 rbconfig.rb       | 2 --
 2 files changed, 4 insertions(+), 2 deletions(-)
 create mode 100644 rbconfig-patch.rb
 delete mode 100644 rbconfig.rb

diff --git a/rbconfig-patch.rb b/rbconfig-patch.rb
new file mode 100644
index 00000000..0a4d372d
--- /dev/null
+++ b/rbconfig-patch.rb
@@ -0,0 +1,4 @@
+require 'rbconfig'
+
+RbConfig::CONFIG["rubyhdrdir"] = File.join("@TERMUX_PREFIX@", "include", "ruby-#{RbConfig::CONFIG['ruby_version']}")
+puts "Using rubyhdrdir=#{RbConfig::CONFIG['rubyhdrdir']}"
diff --git a/rbconfig.rb b/rbconfig.rb
deleted file mode 100644
index 5bf0e54d..00000000
--- a/rbconfig.rb
+++ /dev/null
@@ -1,2 +0,0 @@
-require 'rbconfig'
-puts 'aaaaaaaaaaaaaa'

From f020fd1ac30905cc273c30e9b181e69d1c5f63f0 Mon Sep 17 00:00:00 2001
From: Elliana May <me@mause.me>
Date: Wed, 1 Apr 2026 17:01:05 +0800
Subject: [PATCH] sanity

---
 rbconfig-patch.rb | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/rbconfig-patch.rb b/rbconfig-patch.rb
index 0a4d372d..ade6cc7c 100644
--- a/rbconfig-patch.rb
+++ b/rbconfig-patch.rb
@@ -2,3 +2,6 @@ require 'rbconfig'
 
 RbConfig::CONFIG["rubyhdrdir"] = File.join("@TERMUX_PREFIX@", "include", "ruby-#{RbConfig::CONFIG['ruby_version']}")
 puts "Using rubyhdrdir=#{RbConfig::CONFIG['rubyhdrdir']}"
+if not File.exist?(($hdrdir = RbConfig::CONFIG["rubyhdrdir"]) + "/ruby/ruby.h")
+  abort "#{hdrdir}/ruby/ruby.h does not exist"
+end
